onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -904,6 +904,194 @@ ONNX_MS_OPERATOR_SET_SCHEMA(IsAllFinite, 1,
+      at PyTorch's implementation (https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Col2Im.cpp#L10)
+
+)DOC";
+
+ONNX_MS_OPERATOR_SET_SCHEMA(Col2Im, 1,
+                            OpSchema()
+                            .SetDoc(Col2Im_ver1_doc)
+                            .Attr(
+                                "dilations",
+                                "1-dimensional tensor with dilation value along each spatial axis of the image. "
+                                "If not present, the dilation defaults to 1 along each spatial axis of the image.",
+                                AttributeProto::INTS,
+                                OPTIONAL_VALUE)
+                            .Attr(
+                                "pads",
+                                "1-dimensional tensor with padding value for the beginning and ending along each"
+                                " spatial axis, it can take any value greater than or equal to 0. "
+                                "The value represent the number of pixels added to the beginning "
+                                "and end part of the corresponding axis. `pads` format should be as follow "
+                                "[x1_begin, x2_begin...x1_end, x2_end,...], where xi_begin is the number of pixels "
+                                "added at the beginning of axis `i` and xi_end the same for the end of axis `i`. "
+                                "If not present, the padding defaults to 0 along start and end of each spatial axis.",
+                                AttributeProto::INTS,
+                                OPTIONAL_VALUE)
+                            .Attr(
+                                "strides",
+                                "1-dimensional tensor with stride value along each spatial axis. "
+                                "If not present, the stride defaults to 1 along each spatial axis.",
+                                AttributeProto::INTS,
+                                OPTIONAL_VALUE)
+                            .Input(
+                                0,
+                                "input",
+                                "Input data tensor to be rearranged from column blocks back into an image."
+                                " This is a 3-dimensional tensor containing [N, C * n-ary-product(block_shape), L],"
+                                " where N is batch dimension, C is image channel dimension and L is number of blocks.",
+                                "T",
+                                OpSchema::Single,
+                                true,
+                                1,
+                                OpSchema::Differentiable)
+                            .Input(
+                                1,
+                                "image_shape",
+                                "The shape of the spatial dimensions of the image after rearranging the column blocks."
```

Comment:
nit: no space between sentences. can we be consistent about line breaks, e.g., no leading space and end with a trailing space if the line continues?